Source/Foundation/bsfEngine/Utility/BsShapeMeshes3D.cpp

```diff
@@ -1202,7 +1202,7 @@ namespace bs
 			{
 				int offsetA = i == 0 ? numArcVertices - 1 : i - 1;
 				int offsetB = i;
-				int offsetC = i == numArcVertices ? 1 : i + 1;
+				int offsetC = i == (INT32)numArcVertices ? 1 : i + 1;
```

Comment:
Not entirely sure about this exact fix. Might just want to change the type of `i` to `UINT32`, but that might have implications about conversions to `int` for the offset values. Don't know if we care about that at all?

Reply:
I think changing to `UINT32` should be safe. I'll make the change before I merge.